libcudacxx/include/cuda/std/__cccl/assert.h

```diff
@@ -62,7 +62,7 @@
 //! _CCCL_ASSERT_IMPL_HOST should never be used directly
 #if defined(_CCCL_COMPILER_NVRTC) // There is no host standard library in nvrtc
 #  define _CCCL_ASSERT_IMPL_HOST(expression, message) ((void) 0)
-#elif __has_include(<yvals_core.h>) // MSVC uses _STL_VERIFY from <yvals.h>
+#elif __has_include(<yvals_core.h>) && defined(_CCCL_COMPILER_MSVC) // MSVC uses _STL_VERIFY from <yvals.h>
 #  include <yvals.h>
```

Comment:
The `__has_include` is checking for a different header file, `<yvals_core.h>`, than the header that is actually included, `<yvals.h.>`.  That looks like a bug that should be fixed, even if it wasn't what triggered the NVBug.

Reply:
+1

Reply:
Yeah that was me changing the include

Reply:
STL has both. Headers, clang has only the one, @miscco do we need to check for yvals_core.h for Clang's benefit?

Reply:
No we should only check for the proper MSVC header and I dont mind also checking for MSVC